lambda_calc/wrapper.py

```diff
@@ -0,0 +1,131 @@
+from typing import Generator, TypedDict, TypeAlias
+from functools import reduce
+import itertools
+import string
+
+from .core import alpha_equiv, get_env, all_beta_reductions, is_valid_reduction, is_simple
+from .parser import parse
+from .ast import Var, Fun, App
+
+
+class Reducer:
+    """
+    Wrapper class that holds all the information of the reduced lambda expression
+    
+    lambdastring: Lambda string to evaluate
+    candidatestring: prairelearn input string from candidate/student
+    """
+
+    alpha_error_list = []
+    beta_error_list = []
+    general_error_list = []
+
+    def __init__(self, lambdastring : str, candidatestring : str):
+        self.lambdastring = lambdastring
+        self.candidatestring = candidatestring
+        self.expr = parse(self.lambdastring)
+        self.all_beta_reducs = self.store_all_reducs()
+        self.candidate_beta_reducs = self.parse_candidate_str()
+        #Check the beta reductions
+        self.check_all_beta_reducs()
```

Comment:
this is an opinion but if everything that a class does is happening inside the constructor can't it just be a function? I don't really see why this needs to be a class.

Reply:
I think it should be a class since a lot of variables and expressions are stored. Also, for the Prairelearn check, it can be easily used to provide hints without having to explicitly store variables. The use of OOP might also make the code cleaner and also makes it easier to hook it up with Matteo's code.